package com.kaku.ui.common

import timber.log.Timber
import kotlin.coroutines.cancellation.CancellationException

suspend inline fun <T> suspendRunCatching(block: suspend () -> T): Result<T> = try {
    Result.success(block())
} catch (cancellationException: CancellationException) {
    throw cancellationException
} catch (exception: Exception) {
    Timber.i( "Returning failure Result")
    Result.failure(exception)
}
